src/ui/utils/tokenManager.tsx

```diff
@@ -133,7 +138,10 @@ class TokenManager {
       try {
         const token = await this.fetchToken(refresh);
 
-        this.setState({ token }, deferredState);
+        const decodedToken = jwt_decode<any>(token);
+        const userId = decodedToken?.["https://hasura.io/jwt/claims"]?.["x-hasura-user-id"];
+
+        this.setState({ token, userId }, deferredState);
```

Comment:
I think it might be nice to make this a bit more general. My thinking is that in the future there might be other claims and it is nice for the tokenManager, to not expose arbitrary fields

```
const claims = { hasura: { userId:  decodedToken?.["https://hasura.io/jwt/claims"]?.["x-hasura-user-id"] } }
this.setState({ token, claims }, deferredState);
```

Reply:
I like the idea